src/js/components/Select/doc.js

```diff
@@ -161,3 +161,53 @@ export const doc = Select => {
 
   return DocumentedSelect;
 };
+
+export const themeDoc = {
+  'select.background': {
+    description: 'The background color used for Select.',
+    type: 'string',
+    defaultValue: undefined,
+  },
+  'select.container': {
+    description: 'The container of the Select component.',
+    type: 'string',
+    defaultValue: undefined,
+  },
+  'select.container.extend': {
+    description:
+      'Any additional style for the container of the Select component.',
+    type: 'string | (props) => {}',
+    defaultValue: undefined,
+  },
+  'select.control': {
+    description: 'The control of the Select component.',
+    type: 'string',
+    defaultValue: undefined,
+  },
+  'select.control.extend': {
+    description:
+      'Any additional style for the control of the Select component.',
+    type: 'string | (props) => {}',
+    defaultValue: undefined,
+  },
+  'select.icons.color': {
+    description: 'The color used for Select icons.',
+    type: 'string | { dark: string, light: string }',
+    defaultValue: undefined,
+  },
+  'select.icons.down': {
+    description: 'The down icon to use for opening the Select.',
+    type: 'React.element',
+    defaultValue: '<FormDown />',
+  },
+  'select.searchInput': {
+    description: `The possible padding sizes and the focus indicator of the Select input field.`,
```

Comment:
I think this works as a JSX component. So, you can replace the TextInput used by Select when `onSearch` is provided and instead provide your own. This would set whatever properties you wanted on it.